x/emissions/keeper/inference_synthesis/network_inferences.go

```diff
@@ -314,15 +310,26 @@ func GetCalcNetworkInferenceArgs(
 		InfererToInference:                   infererToInference,
 		InfererToRegret:                      infererToRegret,
 		AllInferersAreNew:                    allInferersAreNew,
-		Forecasters:                          sortedForecasters,
-		ForecasterToForecast:                 forecasterToForecast,
-		ForecasterToRegret:                   forecasterToRegret,
-		ForecasterToForecastImpliedInference: forecastImpliedInferencesByWorker,
+		Forecasters:                          make([]Forecaster, 0),
+		ForecasterToForecast:                 make(map[Forecaster]*emissions.Forecast, 0),
+		ForecasterToRegret:                   make(map[Forecaster]*alloraMath.Dec, 0),
+		ForecasterToForecastImpliedInference: make(map[Forecaster]*emissions.Inference, 0),
 		NetworkCombinedLoss:                  networkLosses.CombinedValue,
 		EpsilonTopic:                         topic.Epsilon,
 		EpsilonSafeDiv:                       moduleParams.EpsilonSafeDiv,
 		PNorm:                                topic.PNorm,
 		CNorm:                                moduleParams.CNorm,
 	}
+
+	// If there are forecast-implied inferences, add forecasters info
+	// It will not have available forecast-implied inferences if the forecasters
+	// didn't make any forecasts for the existing inferers
+	if len(forecastImpliedInferencesByWorker) > 0 {
+		calcArgs.Forecasters = sortedForecasters
+		calcArgs.ForecasterToForecast = forecasterToForecast
+		calcArgs.ForecasterToRegret = forecasterToRegret
+		calcArgs.ForecasterToForecastImpliedInference = forecastImpliedInferencesByWorker
```

Comment:
in case len > 0, is no-nil guaranteed in the other maps? (sortedForecasters, forecasterToForecast, ...)

Reply:
yes!